pkg/reconciler/v1alpha1/taskrun/status/pod.go

```diff
@@ -0,0 +1,226 @@
+package status
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/knative/pkg/apis"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	listers "github.com/tektoncd/pipeline/pkg/client/listers/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/reconciler/v1alpha1/taskrun/resources"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// imageDigestExporterContainerName defines the name of the container that will collect the
+	// built images digest
+	imageDigestExporterContainerName = "step-image-digest-exporter"
+)
+
+// UpdateStatusFromPod modifies the task run status based on the pod and then returns true if the pod is running and
+// all sidecars are ready
+func UpdateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, resourceLister listers.PipelineResourceLister, kubeclient kubernetes.Interface, logger *zap.SugaredLogger) bool {
+	if taskRun.Status.GetCondition(apis.ConditionSucceeded) == nil || taskRun.Status.GetCondition(apis.ConditionSucceeded).Status == corev1.ConditionUnknown {
+		// If the taskRunStatus doesn't exist yet, it's because we just started running
+		taskRun.Status.SetCondition(&apis.Condition{
+			Type:    apis.ConditionSucceeded,
+			Status:  corev1.ConditionUnknown,
+			Reason:  ReasonRunning,
+			Message: ReasonRunning,
+		})
+	}
+
+	taskRun.Status.PodName = pod.Name
+
+	taskRun.Status.Steps = []v1alpha1.StepState{}
+	for _, s := range pod.Status.ContainerStatuses {
+		if resources.IsContainerStep(s.Name) {
+			taskRun.Status.Steps = append(taskRun.Status.Steps, v1alpha1.StepState{
+				ContainerState: *s.State.DeepCopy(),
+				Name:           resources.TrimContainerNamePrefix(s.Name),
+			})
+		}
+	}
+
+	// Complete if we did not find a step that is not complete, or the pod is in a definitely complete phase
+	complete := areStepsComplete(pod) || pod.Status.Phase == corev1.PodSucceeded || pod.Status.Phase == corev1.PodFailed
+
+	if complete {
```

Comment:
nit: This could be splitted once more (aka `if complete { function } else { function }`)

Reply:
nice.  doing this also made me realise that there was some dead code here; the `else` branch was handling pod phases of PodSucceeded and PodFailed but neither of these phases will ever reach the `else` branch since `complete` will be true in both of these cases.